### PR TITLE
Property computation and Analyze panel refactor

### LINKED ISSUE #328: Refactor UI for property computation
As discussed on Dec 2, 2022, should refactor the UI for property computation to streamline and make more intuitive. Here's the Figma with the basic ideas, subject to creative license:
[FIGMA](https://www.figma.com/file/YDwgvMsMWyp2rQxaNfZ3sB/Property-calculation-UI-ideas?node-id=0%3A1&t=b5CY43gi905Ncnmk-0)

First refactor panels:
image

The "Analyze" panel should be some combination of a dialog and a list. At the top, there is a (dynamic) dialog for creating properties, and below there is a list of properties that the user has already added.

Dialog part:
image

List part:
image

It would be nice to get some visual feedback on what got computed after you click "Compute", but not sure how hard that is to do, so it's lower priority for now. We could either refer back to the annotation list, or provide visual feedback like with tooltips.

### PR DESCRIPTION
**Pull request by bruyeret**
See issue #328 for more information on the content of this PR.

Closes #328

### CODE REVIEW

**Comment by bruyeret** [comment]
The workers have to be compiled the same way as before but with labels.
Here is a ts interface showing the different possible labels:
```export interface IWorkerLabels {
  isUPennContrastWorker: string;
  isAnnotationWorker?: string;
  isPropertyWorker?: string;
  interfaceName?: string;
  interfaceCategory?: string;
  annotationShape?: AnnotationShape;
}
```

Each worker needs to have the `isUPennContrastWorker` label defined (no need to give it a value with my changes).
Defining `isAnnotationWorker` or `isPropertyWorker` defines the role of the worker.
A property worker can have an `interfaceName`, an `interfaceCategory` and an `annotationShape`.

For example, to build the blob intensity mean worker from scratch I use this command:
`docker build . -t annotations/blob_intensity_mean_worker:latest -l isUPennContrastWorker -l isPropertyWorker -l "annotationShape=polygon" -l "interfaceName=Blob Intensity Mean" -l "interfaceCategory=Intensity"`

But if you have a worker already built and you want to add the label `foo=bar` (here with the same exemple):
`echo "FROM annotations/blob_intensity_mean_worker:latest" | docker build -l "foo=bar" -t "annotations/blob_intensity_mean_worker:latest"`

**Comment by arjunrajlab** [comment]
When trying to run the above, I get "unknown shorthand flag: 'l' in -l
See 'docker build --help'."

**Comment by bruyeret** [comment]
Oh yes sorry, replace `-l` with `--label`

File: src/components/AnnotationBrowser/AnnotationProperties/Property.vue [hunk 1]
```diff
@@ -1,27 +1,57 @@
...
-        @click.stop="toggleFilter"
-      ></v-checkbox>
-    </v-col>
-    <v-col class="pa-1" cols="7">
-      {{ property.name }}
-    </v-col>
-  </v-row>
+  <v-container class="ma-0 pa-0">
+    <v-row class="mr-4">
+      <!-- In list (???) -->
```
Comment (luciemac) [inline src/components/AnnotationBrowser/AnnotationProperties/Property.vue:4]: To remove ?

File: src/components/AnnotationBrowser/AnnotationProperties/PropertyList.vue [hunk 2]
```diff
@@ -59,40 +99,51 @@ import filterStore from "@/store/filters";
...
+    }
+    return res;
   }
 
-  get layerItems() {
-    return this.layers.map((layer, index) => ({
-      label: layer.name,
-      value: index
-    }));
+  computeUncomputed() {
```
Comment (luciemac) [inline src/components/AnnotationBrowser/AnnotationProperties/PropertyList.vue:137]: computeUncomputed -> computeUncomputedProperties?

File: src/utils/annotation.ts [hunk 3]
```diff
@@ -153,3 +154,16 @@ export function annotationDistance(a: IAnnotation, b: IAnnotation) {
   logError("Unsupported annotation shapes for distance calculations");
   return Number.POSITIVE_INFINITY;
 }
+
+export function canCompute(
```
Comment (luciemac) [inline src/utils/annotation.ts:158]: Can you set a more explicit method name?

**Comment by luciemac** [comment]
Can you add some documentation with all these information?

File: devops/girder/plugins/AnnotationPlugin/README.rst [hunk 4]
```diff
@@ -2,4 +2,22 @@
...
 *******************************
 
-**TODO:** Fill in a description of the project here.
+Workers
+=======
+
+Labels are used to define properties for the worker.
+The typescript interface describing the different possible labels is ``IWorkerLabels`` and can be found in ``src/store/models.ts``.
+
+The following informations may change in the future, but for now:
```
Comment (luciemac) [inline devops/girder/plugins/AnnotationPlugin/README.rst:11]: informations -> information

File: devops/girder/plugins/AnnotationPlugin/upenncontrast_annotation/server/api/propertyValues.py [hunk 5]
```diff
@@ -1,65 +1,80 @@
...
-    .param('propertyId', 'The id of the property')
-    .param('datasetId', 'The id of the dataset')
-    .param('buckets', 'The number of buckets', required=False)
-    )
-    def histogram(self, params):
-        if 'buckets' in params:
-            return self._annotationPropertyValuesModel.histogram(params['propertyId'], params['datasetId'], int(params['buckets']))
-        else:
-            return self._annotationPropertyValuesModel.histogram(params['propertyId'], params['datasetId'])
+from girder.api import access
```
Comment (luciemac) [inline devops/girder/plugins/AnnotationPlugin/upenncontrast_annotation/server/api/propertyValues.py:1]: Seems like file content is the same.

Comment (luciemac) [reply devops/girder/plugins/AnnotationPlugin/upenncontrast_annotation/server/api/propertyValues.py:1]: Isn't it any issue with the end of line character?

File: devops/girder/plugins/AnnotationPlugin/upenncontrast_annotation/server/models/propertyValues.py [hunk 6]
```diff
@@ -1,107 +1,116 @@
...
-        }
-
-        return self.collection.aggregate([
-            match,
-            bucket,
-            project
-        ])
-
-    # def SSE for property change, sends the whole annotation
+from girder.models.model_base import AccessControlledModel
```
Comment (luciemac) [inline devops/girder/plugins/AnnotationPlugin/upenncontrast_annotation/server/models/propertyValues.py:1]: Same here

**Comment by arjunrajlab** [comment]
I tried the following command to build a property worker, but it didn't seem to work properly…
`docker build . -t annotations/blob_area_worker:latest --label "isUPennContrastWorker=True" --label "isPropertyWorker=True" --label "annotationShape=polygon" --label "interfaceName=Blob Area 6" --label "interfaceCategory=Shape"`

**Comment by arjunrajlab** [comment]
Yeah, tried that… still didn't work somehow! All good, I'll try on my Intel box ASAP.

**Comment by luciemac** [comment]
LGTM

**Comment by arjunrajlab** [comment]
Can be merged!

**Pull request merged**